src/main/java/bit/javaoop/Superior.java

```diff
@@ -0,0 +1,41 @@
+package bit.javaoop;
+
+import java.util.List;
+
+public interface Superior {
+    List<Inferior> getInferiors();
+
+    void shouldAddEmployee(Inferior employee);
+
+    default void mergeInferiorList(Superior other)
+    {
+        if(other != null && other.getInferiors().size() != 0)
+        {
+            other.getInferiors().stream()
+                    .filter(x -> !getInferiors().contains(x))
+                    .forEach(y -> getInferiors().add(y));
+        }
+    }
+
+    default String printInferiors(int level)
+    {
+        StringBuilder sb = new StringBuilder("");
+        for (Inferior e: getInferiors())
+        {
+            for (int i = level; i>0; i--)
+            {
+                sb.append("\t");
+            }
+            sb.append(e.getName())
+            .append(" - ")
+            .append(e.getClass().getSimpleName())
+            .append("\n");
+
+            if (e instanceof Superior)
```

Comment:
To też jest bardzo nieładne generalnie instanceof to takie schodzenie z abstrakcją w dół bardzo nieobiektowe i generalnie nie używa się tego. Kiedy pasuje Ci w jakimś miejscu w kodzie instanceof to znak, że prawdopodobnie masz zły design aplikacji.